src/handle/compile.ml

```diff
@@ -88,6 +88,8 @@ let rec compile_with :
       Path.Map.iter compile !(sign.sign_deps);
       loaded := Path.Map.add mp sign !loaded;
       Sign.link sign;
+      if Path.Map.mem Unif_rule.path !(sign.sign_deps) then
```

Comment:
I think this should always be the case since ghost signatures are dependencies of all signatures.

Reply:
Ah right. I should test whether there are rules added. Thanks!

Reply:
Do you have other remarks?

Reply:
You can systematically update the tree: since the operation is lazy, it shouldn't cost more.

Reply:
See https://github.com/Deducteam/lambdapi/pull/722.